server/routes/probationPractitionerReferrals/interventionProgressPresenter.ts

```diff
@@ -152,4 +165,51 @@ export default class InterventionProgressPresenter {
     linkHref: `/probation-practitioner/end-of-service-report/${this.referral.endOfServiceReport?.id}`,
     linkText: 'View',
   }
+
+  private readonly supplierAssessmentSessionStatus = sessionStatus.forAppointment(
+    new SupplierAssessmentDecorator(this.supplierAssessment).currentAppointment
+  )
+
+  readonly supplierAssessmentSessionStatusPresenter = new SessionStatusPresenter(this.supplierAssessmentSessionStatus)
+
+  private get supplierAssessmentStatus(): SupplierAssessmentStatus {
+    if (this.supplierAssessmentSessionStatus !== SessionStatus.notScheduled) {
+      return SupplierAssessmentStatus.scheduled
+    }
+
+    return this.referral.assignedTo === null
+      ? SupplierAssessmentStatus.awaitingCaseworker
+      : SupplierAssessmentStatus.notScheduled
+  }
+
+  readonly shouldDisplaySupplierAssessmentSummaryList =
```

Comment:
I wonder if it might be easier to understand/read/test if we have a method that return SummaryListItem[] and uses ViewUtils.summaryListArgs(), but it will require modifing the SummaryListItem to have a few more fields to handle macro.

Reply:
Yeah, we don't have a general-purpose way for presenters to represent the contents of summary lists. We have `SummaryListItem` which is useful for simple lists where the contents are just text, but it's not clear to me how we might extend it to allow arbitrary contents. Did you have any thoughts on what the `SummaryListItem` interface might look like?